internal/controller/controllers/agent_controller.go

```diff
@@ -211,6 +212,160 @@ func (r *AgentReconciler) Reconcile(origCtx context.Context, req ctrl.Request) (
 	return r.updateStatus(ctx, log, agent, origAgent, &h.Host, h.ClusterID, nil, false)
 }
 
+func splitNamespacedName(nsName string) (string, string, error) {
+	parts := strings.Split(nsName, "/")
+	if len(parts) != 2 {
+		return "", "", fmt.Errorf("unable to parse namespaced name %s", nsName)
+	}
+	return parts[0], parts[1], nil
+}
+
+func deleteBMHForMachine(ctx context.Context, spokeClient client.Client, machine *machinev1beta1.Machine) error {
+	bmhNSName, haveBMH := machine.GetAnnotations()["metal3.io/BareMetalHost"]
+	if !haveBMH {
+		return nil
+	}
+
+	bmhNamespace, bmhName, err := splitNamespacedName(bmhNSName)
+	if err != nil {
+		return errors.Wrap(err, "failed to parse BMH name")
+	}
+
+	bmh := &bmh_v1alpha1.BareMetalHost{}
+	bmhKey := client.ObjectKey{Namespace: bmhNamespace, Name: bmhName}
+	if err = spokeClient.Get(ctx, bmhKey, bmh); err != nil {
+		return errors.Wrapf(err, "failed to get BMH %s", bmhNSName)
+	}
+
+	// TODO: remove this once OCPBUGS-7581 is fixed
```

Comment:
With a fix on the way (https://github.com/metal3-io/baremetal-operator/pull/1259), is there any way we can avoid doing this at all? Or will the timing not work out for that?

Do we have a jira issue to come back and remove this?

Reply:
I'm not sure we'd be able to get that backported this late in the 4.13 cycle.

Even if that PR were merged today it would probably be a week or so before it gets pulled into `openshift/baremetal-operator` then we'd need to do the bug dance to get it into 4.13

I can open a ticket to track removing this.

Reply:
Also just thought of this ...

Should we be doing this update after we delete to ensure BMO doesn't add it back before we can delete it and have it GCd?

Reply:
It might be harder to unit test, but probably worth it to avoid this bug.

Reply:
Created https://issues.redhat.com/browse/MGMT-14495 to track removing the finalizer removal once BMO fixes their bug.

Reply:
I've tested this a lot and haven't seen this. Resolving this conversation for now.